src-gui/commandsender.cpp

```diff
@@ -0,0 +1,121 @@
+#include <QDebug>
+
+#include "commandsender.h"
+#include "sendedcommand.h"
+
+CommandSender::CommandSender(commandExecutor* executor,QObject *parent)
+                : QObject(parent),m_executor(executor),m_counter(QString("fla"))
+{
+    m_sended = new SendedCommands(this);
+    startCommunication();
+}
+
+void CommandSender::startCommunication()
+{
+    auto get_rpc_name = prepareCommand(orderType::GETNAME);
+    m_executor->sendNetRequest(*get_rpc_name);
+    m_wait_name_flag = true;
+}
+
+void CommandSender::onGetName(const QString& client_name)
+{
+    m_client_name = client_name;
+    m_counter.setRpcName(m_client_name);
+    m_wait_name_flag = false;
+}
+
+std::shared_ptr<order> CommandSender::getOrder(const QString &rpc_num)
+{
+    return m_sended->getOrder(rpc_num);
+}
+
+void CommandSender::sendCommand(orderType type, const MeshPeer &peer)
+{
+    if(m_wait_name_flag) {
+        throw std::runtime_error ("can't send msg while witing for rpc name of client");
+    }
+
+    try{
+            auto ord = prepareCommand(type,peer);
+            m_executor->sendNetRequest(*ord);
+            m_sended->addOrder(ord,QString::fromStdString(ord->getId()));
+
+    } catch(std::runtime_error &e) {
+        qDebug()<< "exception while sending :" << e.what();
+    }
+
+}
+
+void CommandSender::sendCommand(CommandSender::orderType type)
+{
+    if(m_wait_name_flag && type != orderType::GETNAME) {
+        throw std::runtime_error ("can't send msg while witing for rpc name of client");
+    }
+
+    try{
+            auto ord = prepareCommand(type);
+            m_executor->sendNetRequest(*ord);
+            m_sended->addOrder(ord,QString::fromStdString(ord->getId()));
+    } catch(std::runtime_error &e) {
+        qDebug()<< "exception while sending :" << e.what();
+    }
+}
+
+std::shared_ptr<order> CommandSender::prepareCommand(CommandSender::orderType type)
+{
+    std::shared_ptr<order> ord;
+
+    switch (type) {
+    case orderType::PING:
+        if(m_client_name.size() == 0)
+            throw std::runtime_error ("no client name provided - can't send command");
+            ord = std::make_shared<pingOrder>( new pingOrder(m_counter.getRpcId()));		//! @todo przerobic na zwaierajaca rpc_num
```

Comment:
all make_shared should be used as:

          ord = std::make_shared<pingOrder>( m_counter.getRpcId());